pubsub/mqtte4k/metadata.go

```diff
@@ -71,20 +75,30 @@ func parseMQTTMetaData(md pubsub.Metadata, log logger.Logger) (*metadata, error)
 		return &m, fmt.Errorf("%s missing url", errorMsgPrefix)
 	}
 
-	if val, ok := md.Properties[spiffeSocketPath]; ok && val != "" {
-		m.spiffeSocketPath = val
-	} else {
-		return &m, fmt.Errorf("%s Invalid or Missing spiffeSocketPath", errorMsgPrefix)
-	}
+	if val, ok := md.Properties[brokerAuthMethod]; ok && val == "spiffe" {
+		m.brokerAuthMethod = "spiffe"
+		if val, ok := md.Properties[spiffeSocketPath]; ok && val != "" {
+			m.spiffeSocketPath = val
+		} else {
+			return &m, fmt.Errorf("%s Invalid or Missing spiffeSocketPath", errorMsgPrefix)
+		}
 
-	if val, ok := md.Properties[spiffeBrokerAudience]; ok && val != "" {
-		m.spiffeBrokerAudience = val
+		if val, ok := md.Properties[spiffeBrokerAudience]; ok && val != "" {
+			m.spiffeBrokerAudience = val
+		} else {
+			return &m, fmt.Errorf("%s Invalid or Missing spiffeBrokerAudience", errorMsgPrefix)
+		}
 	} else {
-		return &m, fmt.Errorf("%s Invalid or Missing spiffeBrokerAudience", errorMsgPrefix)
+		m.brokerAuthMethod = "SAT"
+		if val, ok := md.Properties[satTokenPath]; ok && val != "" {
+			m.satTokenPath = val
+		} else {
+			return &m, fmt.Errorf("%s Invalid or Missing satTokenPath", errorMsgPrefix)
+		}
 	}
 
 	// optional configuration settings
-
+	// make this mandatory
```

Comment:
why?

Reply:
Whoops. This comment was just reminder to me that we always need clientid so the check for it should be mandatory